KsApi/MockService.swift

```diff
@@ -631,17 +675,15 @@
       return SignalProducer(value: CategoryEnvelope(node: .template |> Category.lens.id .~ "\(query.head)"))
     }
 
-    internal func fetchGraphUser(withStoredCards _: Bool)
+    internal func fetchGraphUser(withStoredCards: Bool)
       -> SignalProducer<UserEnvelope<GraphUser>, ErrorEnvelope> {
-      if let error = self.fetchGraphUserResult?.error {
-        return SignalProducer(error: error)
-      } else if let response = self.fetchGraphUserResult?.value {
-        return SignalProducer(value: response)
-      } else {
-        return .empty
-      }
+      let fetchGraphUserQuery = GraphAPI.FetchUserQuery(withStoredCards: withStoredCards)
+
+      return self.apolloClient
+        .fetchWithResult(query: fetchGraphUserQuery, result: self.fetchGraphUserResult)
     }
 
+    // TODO: Refactor this test to use `self.apolloClient`, `ErroredBackingsEnvelope` needs to be `Decodable` and tested in-app.
```

Comment:
Yea so for calls that have this comment, basically I can't use the MockClient like in the calls ala:
`self.apolloClient.performWithResult(mutation: mutation, result: self.changeCurrencyResult)`
because the `performWithResult` and `fetchWithResult` need the result data to be `Decodable`. Right now `ErroredBackingsEnvelope` and `ProjectAndBackingEnvelope` (which is inside `ErroredBackingsEnvelope`) is not `Decodable`. It may as simple as adding `: Decodable` to the Envelopes but they depend on pretty large objects, namely `Project` and `Backing`. So it's a big regression test to ensure those objects are still being passed around correctly with `Decdoable` added to them. Also their properties will need that conformance, so just left those two Envelopes untested. We can return to them when `Project` is exclusively based on `Apollo` properties

Reply:
Ah I was under the impression `Project` and `Backing` both conformed to the protocol already.